KnowYourEnemyMutagen/Program.cs

```diff
@@ -39,315 +29,219 @@ public static int Main(string[] args)
-                perks.Add("rocky", rockyPerk.DeepCopy());
-            else Console.WriteLine("Failed to add perk rocky");
+        private static float AdjustMagicResist(float magnitude, float scale)
+        {
+            return magnitude == 0 ? magnitude : magnitude * scale;
+        }
 
+        private static readonly Tuple<string, uint>[] PerkArray = {
+            new Tuple<string, uint>("fat", 0x00AA5E),
+            new Tuple<string, uint>("big", 0x00AA60),
+            new Tuple<string, uint>("small", 0x00AA61),
+            new Tuple<string, uint>("armored", 0x00AA62),
+            new Tuple<string, uint>("undead", 0x00AA6),
+            new Tuple<string, uint>("plant", 0x00AA6),
+            new Tuple<string, uint>("skeletal", 0x00AA6),
+            new Tuple<string, uint>("brittle", 0x00AA6),
+            new Tuple<string, uint>("dwarven machine", 0x00AA6),
+            new Tuple<string, uint>("ghostly", 0x02E17),
+            new Tuple<string, uint>("furred", 0x04768),
+            new Tuple<string, uint>("supernatural", 0x04768),
+            new Tuple<string, uint>("venomous", 0x04768),
+            new Tuple<string, uint>("ice elemental", 0x04768),
+            new Tuple<string, uint>("fire elemental", 0x04768),
+            new Tuple<string, uint>("shock elemental", 0x04768),
+            new Tuple<string, uint>("vile", 0x04768),
+            new Tuple<string, uint>("troll kin", 0x04768),
+            new Tuple<string, uint>("weak willed", 0x04768),
+            new Tuple<string, uint>("strong willed", 0x04768),
+            new Tuple<string, uint>("cave dwelling", 0x04768),
+            new Tuple<string, uint>("vascular", 0x04768),
+            new Tuple<string, uint>("aquatic", 0x04768),
+            new Tuple<string, uint>("rocky", 0x04C78E)
+        };
 
-            foreach (KeyValuePair<string, Perk> entry in perks)
+        private static IEnumerable<string> GetFromJson(string key, JObject jObject)
+        {
+            return jObject.ContainsKey(key) ? jObject[key]!.Select(x => (string?) x).Where(x => x != null).Select(x => x!).ToList() : new List<string>();
+        }
+        
+        private static void RunPatch(SynthesisState<ISkyrimMod, ISkyrimModGetter> state)
+        {
+            if (!state.LoadOrder.ContainsKey(ModKey.FromNameAndExtension("know_your_enemy.esp")))
             {
-                Console.WriteLine(entry.Key + " is Key");
-                /*foreach (string k in entry.Key)
-                {
-                    Console.WriteLine(k);
-                    //Console.WriteLine(val + " belongs to it");
-                }
-                */
+                Console.WriteLine("ERROR: Know Your Enemy not detected in load order. You need to install KYE prior to running this patcher!");
+                return;
             }
 
-
-            // ***** Part 0 *****
```

Comment:
Keeping the part comments, they're in the original so I think it makes it a bit easier to understand where each part comes in.